int32 id
time start
int32 msgs_left
string data
